pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,608 @@
+	if err != nil {
+		return nil, fmt.Errorf("compose neo-go server config from the base one: %w", err)
+	}
+
+	blockchainStorage, err := storage.NewStore(cfg.ApplicationConfiguration.DBConfiguration)
+	if err != nil {
+		return nil, fmt.Errorf("init blockchain storage: %w", err)
+	}
+
+	blockChain, err := core.NewBlockchain(blockchainStorage, cfg.Blockchain(), log)
+	if err != nil {
+		closeErr := blockchainStorage.Close()
+		if closeErr != nil {
+			return nil, fmt.Errorf("init core blockchain component: %w; failed to close blockchain storage: %v", err, closeErr)
+		}
+
+		return nil, fmt.Errorf("init core blockchain component: %w", err)
+	}
+
+	go blockChain.Run() // is it required to be insta called?
+
+	netServer, err := network.NewServer(cfgServer, blockChain, blockChain.GetStateSyncModule(), log)
+	if err != nil {
+		return nil, fmt.Errorf("init neo-go network server: %w", err)
+	}
+
+	rpcActor := newActor(blockChain, netServer, cfg.ApplicationConfiguration.RPC.MaxGasInvoke)
+
+	_actor, err := actor.NewSimple(rpcActor, acc)
+	if err != nil {
+		return nil, fmt.Errorf("init simple actor using node account: %w", err)
+	}
+
+	stateRootMode := blockChain.GetStateModule().(*corestateroot.Module)
+
+	stateRootService, err := stateroot.New(cfgServer.StateRootCfg, stateRootMode, log, blockChain, netServer.BroadcastExtensible)
+	if err != nil {
+		return nil, fmt.Errorf("init StateRoot service: %w", err)
+	}
+
+	netServer.AddExtensibleService(stateRootService, stateroot.Category, stateRootService.OnPayload)
+
+	// do we need oracle service?
+
+	// var oracleService *oracle.Oracle
+	//
+	// if cfg.ApplicationConfiguration.Oracle.Enabled {
+	// 	var c oracle.Config
+	// 	c.Log = log
+	// 	c.Network = cfg.ProtocolConfiguration.Magic
+	// 	c.MainCfg = cfg.ApplicationConfiguration.Oracle
+	// 	c.Chain = artifacts.blockchain
+	// 	c.OnTransaction = artifacts.netServer.RelayTxn
+	//
+	// 	oracleService, err = oracle.NewOracle(c)
+	// 	if err != nil {
+	// 		return fmt.Errorf("init Oracle module: %w", err)
+	// 	}
+	//
+	// 	artifacts.blockchain.SetOracle(oracleService)
+	// 	artifacts.netServer.AddService(oracleService)
+	// }
+
+	var consensusService consensus.Service
+
+	if cfg.ApplicationConfiguration.Consensus.Enabled {
+		var c consensus.Config
+		c.Logger = log
+		c.Broadcast = netServer.BroadcastExtensible
+		c.Chain = blockChain
+		c.ProtocolConfiguration = blockChain.GetConfig().ProtocolConfiguration
+		c.RequestTx = netServer.RequestTx
+		c.StopTxFlow = netServer.StopTxFlow
+		c.Wallet = cfg.ApplicationConfiguration.Consensus.UnlockWallet
+		c.TimePerBlock = cfgServer.TimePerBlock
+
+		consensusService, err = consensus.NewService(c)
+		if err != nil {
+			return nil, fmt.Errorf("init Consensus module: %w", err)
+		}
+
+		netServer.AddConsensusService(consensusService, consensusService.OnPayload, consensusService.OnTransaction)
+	}
+
+	var notaryService *notarysvc.Notary
+
+	if cfg.ApplicationConfiguration.P2PNotary.Enabled {
```

Comment:
This one also doesn't need to be configured explicitly, CNs usually serve as notaries as well.